Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarCloudWebServerTest.cs

```diff
@@ -376,95 +377,100 @@ public async Task DownloadRules_SonarCloud()
+
+        var compressed = new MemoryStream();
+        using var compressor = new GZipStream(compressed, CompressionMode.Compress, true);
+        stream.CopyTo(compressor);
+
+        compressor.Close();
+        compressed.Seek(0, SeekOrigin.Begin);
+        return compressed;
+    }
+
+    private static IDownloader MockIDownloader(string cacheBaseUrl = null)
+    {
+        var serverSettingsJson = cacheBaseUrl is not null
+                                     ? $"{{\"settings\":[{{ \"key\":\"sonar.sensor.cache.baseUrl\",\"value\": \"{cacheBaseUrl}\" }}]}}"
+                                     : "{\"settings\":[]}";
+
+        var mock = Substitute.For<IDownloader>();
+        mock.Download(Arg.Any<string>(), Arg.Any<bool>()).Returns(Task.FromResult(serverSettingsJson));
+        mock.TryDownloadIfExists(Arg.Any<string>(), Arg.Any<bool>()).Returns(Task.FromResult(new Tuple<bool, string>(false, string.Empty)));
+        return mock;
+    }
+
+    private static HttpMessageHandler MockHttpHandler(string cacheFullUrl, string prepareReadResponse, HttpStatusCode prepareReadResponseCode = HttpStatusCode.OK)
+    {
+        var handler = Substitute.For<HttpMessageHandler>();
+        handler.Protected("SendAsync", Arg.Is<HttpRequestMessage>(x => x.RequestUri == new Uri(cacheFullUrl) && x.Headers.Any(h => h.Key == "Authorization" && h.Value.Contains($"Bearer {Token}"))), Arg.Any<CancellationToken>())
+            .Returns(Task.FromResult(new HttpResponseMessage
+            {
+                StatusCode = prepareReadResponseCode,
+                Content = new StringContent(prepareReadResponse),
+            }));
+        return handler;
+    }
+
+    private static HttpMessageHandler MockHttpHandler(string cacheFullUrl, string ephemeralCacheUrl, Stream cacheData)
+    {
+        var mock = MockHttpHandler(cacheFullUrl, $"{{ \"enabled\": \"true\", \"url\":\"{ephemeralCacheUrl}\" }}");
+        mock.Protected("SendAsync", Arg.Is<HttpRequestMessage>(x => x.RequestUri == new Uri(ephemeralCacheUrl)), Arg.Any<CancellationToken>())
+            .Returns(Task.FromResult(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StreamContent(cacheData),
+            }));
+        return mock;
+    }
+
+    private static ProcessedArgs CreateLocalSettings(string projectKey,
+                                                     string branch,
+                                                     string organization = "placeholder",
+                                                     string token = "placeholder",
+                                                     string tokenKey = SonarProperties.SonarToken)
+    {
+        var args = Substitute.For<ProcessedArgs>();
+        args.ProjectKey.Returns(projectKey);
+        args.Organization.Returns(organization);
+        args.TryGetSetting(Arg.Is<string>(x => x == SonarProperties.PullRequestBase), out Arg.Any<string>())
```

Comment:
This looks like an overkill
```suggestion
        args.TryGetSetting(SonarProperties.PullRequestBase, out Arg.Any<string>())
```

Reply:
Same below

Reply:
And above in MockHttpHandler (that one will go away)